cmd/collector/app/server/http.go

```diff
@@ -68,7 +68,7 @@ func StartHTTPServer(params *HTTPServerParams) (*http.Server, error) {
 		if err != nil {
 			return nil, err
 		}
-		server.TLSConfig = tlsCfg
+		server.TLSConfig.GetCertificate = tlsCfg.GetCertificate
```

Comment:
1. when the whole `tlsCfg` struct was assigned, this field was also already assigned, so what is the difference?
2. how do we know this actually fixes the problem?